Documentation/coding-guidelines/project-guidelines.md

```diff
@@ -72,103 +106,40 @@ Project configurations that are unique for a few different target frameworks and
-```xml
-<?xml version="1.0" encoding="utf-8"?>
-<Project ToolsVersion="12.0" DefaultTargets="Build" xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
-  <Import Project="$([MSBuild]::GetDirectoryNameOfFileAbove($(MSBuildThisFileDirectory), dir.props))\dir.props" />
-  <ItemGroup>
-    <Project Include="System.IO.FileSystem.csproj">
-      <OSGroup>FreeBSD</OSGroup>
-    </Project>
-    <Project Include="System.IO.FileSystem.csproj">
-      <OSGroup>Linux</OSGroup>
-    </Project>
-    <Project Include="System.IO.FileSystem.csproj">
-      <OSGroup>OSX</OSGroup>
-    </Project>
-    <Project Include="System.IO.FileSystem.csproj">
-      <OSGroup>Windows_NT</OSGroup>
-    </Project>
-    <Project Include="System.IO.FileSystem.csproj">
-      <OSGroup>Windows_NT</OSGroup>
-      <TargetGroup>uap101</TargetGroup>
-    </Project>
-  </ItemGroup>
-  <Import Project="$([MSBuild]::GetDirectoryNameOfFileAbove($(MSBuildThisFileDirectory), dir.traversal.targets))\dir.traversal.targets" />
-</Project>
+#Library project guidelines
+Library projects should use the following directory layout.
 
 ```
-Project configurations that are unique for a few different target frameworks and runtimes
-```xml
-<?xml version="1.0" encoding="utf-8"?>
-<Project ToolsVersion="12.0" DefaultTargets="Build" xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
-  <Import Project="$([MSBuild]::GetDirectoryNameOfFileAbove($(MSBuildThisFileDirectory), dir.props))\dir.props" />
-  <ItemGroup>
-    <Project Include="System.Linq.Expressions.csproj" />
-    <Project Include="System.Linq.Expressions.csproj">
-      <OSGroup>Windows_NT</OSGroup>
-      <TargetGroup>uap101</TargetGroup>
-    </Project>
-    <Project Include="System.Linq.Expressions.csproj">
-      <OSGroup>Windows_NT</OSGroup>
-      <TargetGroup>uap101aot</TargetGroup>
-    </Project>
-  </ItemGroup>
-  <Import Project="$([MSBuild]::GetDirectoryNameOfFileAbove($(MSBuildThisFileDirectory), dir.traversal.targets))\dir.traversal.targets" />
-</Project>
+src\<Library Name>\src - Contains the source code for the library.
+src\<Library Name>\ref - Contains any reference assembly projects for the library
+src\<Library Name>\pkg - Contains package projects for the library.
+src\<Library Name>\tests - Contains the test code for a library
 ```
+##ref
+Reference assemblies are required for any library that has more than one implementation or uses a facade. A reference assembly is a surface-area-only assembly that represents the public API of the library. To generate a reference assembly source file you can use the [GenAPI tool](https://www.nuget.org/packages/Microsoft.DotNet.BuildTools.GenAPI). If a library is a pure portable library with a single implementation it need not use a reference assembly at all.
 
-###Tests project .builds files
+In the ref directory for the library there should be at most **one** `.csproj` that contains the latest API for the reference assembly for the library.
```

Comment:
Looks like you are not mentioning directly about the cases where we have Api that is only supported in netcoreapp and not as the part of standard. Is that intentional?